### PR TITLE
Version for JDK 1.7/1.8

### PR DESCRIPTION
**Pull request by anlun**
It doesn't compile for me without these changes with JDK 1.7/1.8.

### CODE REVIEW

**Comment by anlun** [comment]
May be the problem is that I'm using JDK 1.7, but I'm not quite sure.

File: plugin/src/org/jetbrains/cabal/psi/VersionConstraint.kt [hunk 1]
```diff
@@ -61,15 +61,14 @@ public class VersionConstraint(node: ASTNode) : ASTWrapperPsiElement(node), Chec
...
 
-    public fun getVersion(): String = (this : PsiElement).getLastChild()!!.getText()!!
+    public fun getVersion(): String = this.getLastChild()!!.getText()!!
 
     public fun getVersionValue(): String = (getVersion().replace("(\\-[0-9a-zA-Z]+)+\\.".toRegex(), "\\.")).replace("(\\-[0-9a-zA-Z]+)+$".toRegex(), "")
 
     public override fun check(): List<ErrorMessage> {
-        val comparator = getComparator()
-        if (comparator == null) return listOf()
+        val comparator = getComparator() ?: return listOf()
```
Comment (superbobry) [inline plugin/src/org/jetbrains/cabal/psi/VersionConstraint.kt]: This seems to be an equivalent expression. Are you sure the previous one caused a compilation error?

**Comment by superbobry** [comment]
Can you please move cosmetic changes (like Kotlin API migrations) into a separate PR, so that we can see the code breaking the compilation?

File: plugin/src/org/jetbrains/cabal/psi/VersionConstraint.kt (hunk 1 above)
Comment (anlun) [inline plugin/src/org/jetbrains/cabal/psi/VersionConstraint.kt]: It's just cosmetic one.

**Comment by anlun** [comment]
Yes, I'll do it in a minute.

**Pull request closed without merging**